common/types/duration.go

```diff
@@ -103,8 +105,22 @@ func (d Duration) Compare(other ref.Val) ref.Val {
 
 // ConvertToNative implements ref.Val.ConvertToNative.
 func (d Duration) ConvertToNative(typeDesc reflect.Type) (interface{}, error) {
-	if typeDesc == durationValueType {
+	switch typeDesc {
+	case anyValueType:
+		return ptypes.MarshalAny(d.Value().(*dpb.Duration))
+	case durationValueType:
 		return d.Value(), nil
+	case jsonValueType:
+		// proto3 to JSON conversion requires string-formatted durations
```

Comment:
Change to "CEL follows the proto3 to JSON conversion."  Consider using jsonpb to render the string, though that slightly mixes the levels.

Note that we can't just use jsonpb to render the string because it would add extra double quotes.